app/cli/internal/action/attestation_init.go

```diff
@@ -242,6 +220,96 @@ func (action *AttestationInit) Run(ctx context.Context, opts *AttestationInitRun
 	return attestationID, nil
 }
 
+// createWorkflow creates a new workflow if it doesn't exist, as well as its contract
+func (action *AttestationInit) createWorkflow(ctx context.Context, opts *AttestationInitRunOpts) (*WorkflowItem, error) {
+	// 1. find workflow if exists
+	wf, err := NewWorkflowDescribe(action.ActionsOpts).Run(ctx, opts.WorkflowName, opts.ProjectName)
+	if err != nil && status.Code(err) != codes.NotFound {
+		return nil, fmt.Errorf("error looking for workflow %q: %w", opts.WorkflowName, err)
+	}
```

Comment:
to simplify readability here I'd `return wf` if wf != nil && contractRef == ""

to indicate there is nothing to create/update